packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -941,9 +1004,15 @@ class _RenderSegmentedButton<T> extends RenderBox with
         final BorderSide divider = segments[index - 1].enabled || segments[index].enabled
           ? enabledBorder.side.copyWith(strokeAlign: 0.0)
           : disabledBorder.side.copyWith(strokeAlign: 0.0);
-        final Offset top = Offset(dividerPos, borderRect.top);
-        final Offset bottom = Offset(dividerPos, borderRect.bottom);
-        context.canvas.drawLine(top, bottom, divider.toPaint());
+        if (direction == Axis.horizontal) {
+          final Offset top = Offset(dividerPos, borderRect.top);
+          final Offset bottom = Offset(dividerPos, borderRect.bottom);
+          context.canvas.drawLine(top, bottom, divider.toPaint());
+        } else if (direction == Axis.vertical) {
+          final Offset start = Offset(borderRect.left, childRect.top);
+          final Offset end = Offset(borderRect.right, childRect.top);
```

Comment:
Should both start.y and end.y be `childRect.bottom`?

Reply:
Yup , this part is responsible for divisions lines, and currently its correct.
Its get draw by the the start of (topLeft of the (element.index-1)) to (topRight of the (element.index-1))
Did i get you right ?

Reply:
I see. I think we also need to consider to clip the divider here. It seems when it's a rounded rectangle segmented buttons, the divider might exceed the border.

<img width="300" src="https://github.com/user-attachments/assets/4a943048-bf27-468e-9a31-062a123fb4da">

Reply:
hi @QuncCccccc , do you have any suggestion to do it ?

Reply:
Yeeeeees i solve it @QuncCccccc  , I check out flutter team code pattern , and i figure out how to do it, you can check it, let me know if you have any feedback's.
![Simulator Screenshot - iPhone 15 - 2024-09-05 at 14 53 21](https://github.com/user-attachments/assets/29377166-4d14-4b42-958e-9e4e5d081af6)

Reply:
NICE! Taking a look:)